packages/cadl-typescript/CONTRIBUTING.md

```diff
@@ -0,0 +1,121 @@
+
+5. You can run the Smoke tests using the following command:
+
+```
+cd ../../packages/cadl-rlc-test
+npm run smoke-test
+```
+
+## How to add an integration test case
+
+Whenever you work on adding a feature/fixing a bug, this would probably be your first step. You create a test case and then run it through the generator, see the result, modify the generator, run it again and so on, until you get the desired output.
+
+1. Pick up a cadl as your test input in cadl-ranch. Below are some examples
+
+Let us say your test input will be called `testUserCase.json`.
+
+2. Now add an entry to the CadlRanchConfig to the file [`cadl-ranch-list.ts`](./test/commands/cadl-ranch-list.ts). In the file, add the following to the array.
+
+```
+  {
+    outputPath: "authentication/apiKey",
+    inputPath: "authentication/api-key"
+  },
+```
+
+3. Now, You can generate the RLC for your test case with the following command: (Initially, during your development, you do not want to run all the cases during every step of your development, you can comment out other test cases. But, once your code changes are complete for your case, then you need to run the entire suite to ensure that your changes did not cause any unwanted changes.)
+
+```
+npm run generate-cadl-only
+```
+
+4. Once you are satisfied with the generated code, you can add a spec file such as `testUserCaseRest.spec.ts` file [here](./test/integration). You can find several examples in the same place.
+
+## How to debug
+
+### `generate-cadl-only` step
+
+If you would like to debug the `generate-cadl-only` step for our test input, use the following command:
+
+```
+npm run generate-cadl-only -- -b -i testUserCase --debug
+```
+
+Now, the code will wait for the debugger to be attached. Open the repository in VS Code -> Select `Run and Debug` section -> Click `Attach`.
+
+### Spec file
+
+If you would like to debug the `testUserCase.spec.ts` file (after the SDK is generated), Open the repository in VS Code -> Open the `testUserCase.spec.ts` file -> Select `Run and Debug` section -> Click `IntegrationTests - Current File`.
+
+### How to debug an unit test case
+
+- In VS Code, We have created a Debugging profile for UnitTests to start debugging:
+
+  1. Go to the debugger tab
+  2. Select the "Unit Test" Profile
+  3. Click the "Play" button
+
+- Your breakpoints will start hitting, you can set breakpoints in either Test or Generator code
+
+### Integration Tests
+
+- In order to debug integration tests you need to start the test server, by running:
+
+      npm run start-test-server:v1
+
+- Once the Test Server is running
+
+  1. In VSCode go to the debugger tab
+  2. Select the "IntegrationTests" profile from the drop down
```

Comment:
Do you mean we could debug cadl UT by config `Cadl Unit or Integration Test`?

I would suggest to use below for Integration test:

```
{
      "name": "[CADL] - Integration Test",
      "request": "launch",
      "cwd": "${workspaceFolder}/packages/cadl-typescript",
      "runtimeArgs": [
        "mocha",
        "-r",
        "ts-node/register",
        "--experimental-specifier-resolution=node",
        "--experimental-modules=true",
        "--timeout",
        "999999",
        "--colors",
        "./test/integration/*.spec.ts"
      ],
      "runtimeExecutable": "npx",
      "skipFiles": ["<node_internals>/**"],
      "type": "pwa-node",
      "outFiles": ["${workspaceFolder}/packages/*/dist/**/*.js"]
    }
```

Reply:
updated.